src/main/java/com/crowdin/client/Crowdin.java

```diff
@@ -162,6 +200,34 @@ public File downloadTranslations(VirtualFile sourceFile, String branch) {
         }
     }
 
+    public Map<Long, Directory> getDirectories(Long branchId) {
+        try {
+            return this.client.getSourceFilesApi()
+                .listDirectories(this.projectId, null, null, null, 500, 0)
+                .getData()
+                .stream()
+                .map(ResponseObject::getData)
+                .filter(dir -> Objects.equals(dir.getBranchId(), branchId))
```

Comment:
Why not pass to end point branch id? Second argument of `listDirectories` method is a branchId which will be used as a filter on a server side.

Reply:
Yes, but they will be filtered only if branchId is not empty. With an empty branchId, all files associated with other branches will also be returned.

Reply:
Oh, and I forgot to add branchId to request, thanks